crates/biome_js_analyze/src/lint/nursery/use_explicit_type.rs

```diff
@@ -550,3 +652,32 @@ fn is_type_assertion(syntax: &SyntaxNode<JsLanguage>) -> bool {
         }
     })
 }
+
+/// Checks if all parameters in a function have explicit type annotations.
+///
+/// This function iterates over the provided function parameters and examines each one
+/// to determine if it has a type annotation. If any parameter is missing a type annotation,
+/// the function returns a `Some(UseExplicitTypeState)` indicating the range of the parameter
+/// and the cause of the issue. If all parameters have type annotations, it returns `None`.
+///
+/// # Arguments
+///
+/// * `parameters` - A reference to a `JsParameters` object representing the parameters of a function.
+///
+/// # Returns
+///
+/// * `Option<UseExplicitTypeState>` - `Some(UseExplicitTypeState)` if a parameter is missing a type annotation,
+///   or `None` if all parameters have explicit type annotations.
+fn check_function_parameters_type(parameters: &JsParameters) -> Option<UseExplicitTypeState> {
+    for p in parameters.items() {
+        let param = p.ok()?;
+        let formal_param = param.as_any_js_formal_parameter()?;
```

Comment:
We should certainly handle the other parameter kinds (rest parameter and `this` parameter).